src/HotChocolate/ApolloFederation/src/ApolloFederation/Configuration/SchemaPrinter/FederationSchemaPrinter.cs

```diff
@@ -126,13 +126,35 @@ private static IEnumerable<INamedType> GetRelevantTypes(ISchema schema)
         return new NamedTypeNode(null, new NameNode(namedType.Name));
     }
 
-    private static IReadOnlyList<DirectiveNode> SerializeDirectives(
+    internal struct MaybeList<T>
```

Comment:
This can be moved to utilities ... we can use this in other parts ... leave it internal though